somaticseq/somatic_vcf2tsv.py

```diff
@@ -0,0 +1,1390 @@
+    inputParameters['varscan']    = args.varscan_vcf
+    inputParameters['jsm']        = args.jsm_vcf
+    inputParameters['sniper']     = args.somaticsniper_vcf
+    inputParameters['vardict']    = args.vardict_vcf
+    inputParameters['muse']       = args.muse_vcf
+    inputParameters['lofreq']     = args.lofreq_vcf
+    inputParameters['scalpel']    = args.scalpel_vcf
+    inputParameters['strelka']    = args.strelka_vcf
+    inputParameters['tnscope']    = args.tnscope_vcf
+    
+    inputParameters['ref']        = args.genome_reference
+    inputParameters['dedup']      = args.deduplicate
+    
+    inputParameters['min_mq']     = args.minimum_mapping_quality
+    inputParameters['min_bq']     = args.minimum_base_quality
+    inputParameters['min_caller'] = args.minimum_num_callers
+    inputParameters['ref_fa']     = args.genome_reference
+    inputParameters['p_scale']    = args.p_scale
+    
+    inputParameters['outfile']    = args.output_tsv_file
+        
+    return inputParameters
+
+
+
+def vcf2tsv(is_vcf=None, is_bed=None, is_pos=None, nbam_fn=None, tbam_fn=None, truth=None, cosmic=None, dbsnp=None, mutect=None, varscan=None, jsm=None, sniper=None, vardict=None, muse=None, lofreq=None, scalpel=None, strelka=None, tnscope=None, ref=None, dedup=True, min_mq=1, min_bq=5, min_caller=0, ref_fa=None, p_scale=None, outfile=None):
+
+    # Convert contig_sequence to chrom_seq dict:
+    fai_file  = ref_fa + '.fai'
+    chrom_seq = genome.faiordict2contigorder(fai_file, 'fai')
+    
+    # Determine input format:
+    if is_vcf:
+        mysites = is_vcf
+    elif is_bed:
+        mysites = is_bed
+    elif is_pos:
+        mysites = is_pos
+    else:
+        mysites = fai_file
+        print('No position supplied. Will evaluate the whole genome.', file=sys.stderr)
+    
+    # Re-scale output or not:
+    if p_scale == None:
+        print('NO RE-SCALING', file=sys.stderr)
+    elif p_scale.lower() == 'phred':
+        p_scale = 'phred'
+    elif p_scale.lower() == 'fraction':
+        p_scale = 'fraction'
+    else:
+        p_scale = None
+        print('NO RE-SCALING', file=sys.stderr)
+    
+        # Define NaN and Inf:
+    nan = float('nan')
+    inf = float('inf')
+    pattern_chr_position = genome.pattern_chr_position
+    
+    # Normal/Tumor index in the Merged VCF file, or any other VCF file that puts NORMAL first. 
+    idxN,idxT = 0,1
+    
+    # Normal/Tumor index in VarDict VCF, or any other VCF file that puts TUMOR first.
+    vdT,vdN = 0,1
+    
+    ## Running
+    with genome.open_textfile(mysites) as my_sites, open(outfile, 'w') as outhandle:
+                
+        my_line = my_sites.readline().rstrip()
+        
+        nbam    = pysam.AlignmentFile(nbam_fn, reference_filename=ref_fa)
+        tbam    = pysam.AlignmentFile(tbam_fn, reference_filename=ref_fa)
+        ref_fa  = pysam.FastaFile(ref_fa)
+        
+        if truth:
+            truth = genome.open_textfile(truth)
```

Comment:
Skipping the header lines is a repetitive operation here for the different VCFs. Should factor it out as a function. This function is too large so would be good to factor some functions out of this to help with readability and extension.